packages/core/src/LottieView/index.web.tsx

```diff
@@ -24,151 +20,128 @@ const LottieView = forwardRef(
       hover,
       direction,
       progress,
+      onAnimationLoaded,
       onAnimationFailure,
       onAnimationFinish,
       onAnimationLoop,
     }: LottieViewProps,
-    ref: any
+    ref: any,
   ) => {
+    const [dotLottie, setDotLottie] = useState<DotLottie>(null);
     const sources = parsePossibleSources(source);
-    const lottieSource = sources?.sourceDotLottieURI || sources?.sourceName;
-    const jsonSource = sources?.sourceURL || sources?.sourceJson;
+    const dotLottieRefCallback = useCallback((dotLottie) => {
+      setDotLottie(dotLottie);
+    }, []);
 
-    const [isError, setIsError] = React.useState(false);
-    const [isReady, setIsReady] = React.useState(false);
-    const [key, setKey] = React.useState(0);
-
-    const playerRef = useRef<DotLottieCommonPlayer | null>(null);
-    /**
-     *  If an error occured reset the key when the source changes to force a re-render.
-     */
     useEffect(() => {
-      if (isError) {
-        setKey((prevKey) => prevKey + 1);
-        setIsError(false);
-      }
-    }, [source]);
+      if (dotLottie) {
```

Comment:
Is this because dotLottie can be not installed, or is this a thing that this package needs by default?

Reply:
Read the documentation. Very strange way of adding event handlers but ok

Reply:
It's because `dotLottie` can be null when this effect runs, particularly when the `sources` prop is not provided.

Reply:
Yeah they force us to use the web api via the ref, would have been nice if the library handled that for us via props but alas. Still cleaner than the previous implementation!